File: gestao-spring/src/main/java/com/erp/gestao/model/Produto.java
package com.erp.gestao.model;

import com.erp.gestao.enums.StatusProdutoEnum;
import com.erp.gestao.utils.validate.ValidateField;
import com.erp.gestao.utils.validate.ValidationType;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name ="produto")
@Data
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @ValidateField(value = {ValidationType.NOT_NULL}, fieldName = "Categoria")
    private Categoria categoria;

    @ManyToOne
    private Marca marca;

    @ValidateField(value = {ValidationType.NOT_NULL}, fieldName = "Nome")
    private String nome;

    @ValidateField(value = {ValidationType.NOT_NULL, ValidationType.UNIQUE}, fieldName = "Codigo de Patrimonio")
    private String codigoPatrimonio;

    private String modelo;

    @Column(precision = 10, scale = 2)
    @ValidateField(value = {ValidationType.NOT_NULL}, fieldName = "Valor da Diaria")
    private BigDecimal valorDiaria;

    @Enumerated(EnumType.STRING)
    private StatusProdutoEnum statusAtual;

    private LocalDate dataAquisicao;

    @Column(columnDefinition = "TEXT")
    private String observacoes;

    @Version
    private Integer versao = 0;

}

File: gestao-angular/src/app/shared/model/produto.ts
import {GridColumn} from "../utils/directives/grid.column.decorator";
import {StatusProdutoEnum} from "../enum/status.produto.enum";
import {Marca} from "./marca";
import {Categoria} from "./categoria";

export class Produto{

  id: number;

  categoria: Categoria;

  marca: Marca;

  @GridColumn({ label: 'Nome', type: 'texto', ordem: 2 })
  nome: string;

  @GridColumn({ label: 'Cod Patriminio', type: 'texto', ordem: 1 })
  codigoPatrimonio: string;

  @GridColumn({ label: 'Modelo', type: 'texto', ordem: 3 })
  modelo: string;

  @GridColumn({ label: 'Valor Diaria', type: 'currency', ordem: 4 })
  valorDiaria: number;

  @GridColumn({ label: 'Status Atual', type: 'enum', ordem: 6 })
  statusAtual: StatusProdutoEnum;

  @GridColumn({ label: 'Data Aquisição', type: 'date', ordem: 5 })
  dataAquisicao: Date;

  observacoes: string

  versao: number;

}
